src/components/proposals/quest-progression.vue

```diff
@@ -0,0 +1,154 @@
+<script>
+import { PROPOSAL_STATE } from '~/const'
+import { mapGetters } from 'vuex'
+import { dateToStringShort } from '~/utils/TimeUtils'
+
+export default {
+  name: 'quest-progression',
+  props: {
+    proposalId: String
+  },
+
+  data () {
+    return {
+      PROPOSAL_STATE
+    }
+  },
+
+  apollo: {
+    progression: {
+      query: require('~/query/quests/dao-quest-progression.gql'),
+
+      update: data => {
+        const currentQuest = data.queryQueststart[0]
+
+        return {
+          currentQuest: {
+            status: currentQuest?.details_state_s,
+            rejectedDate: currentQuest?.updatedDate,
+            approvedDate: currentQuest?.system_originalApprovedDate_t,
+            title: currentQuest?.details_title_s
+          },
+          ascendants: currentQuest?.ascendant.map(ascendant => {
+            return {
+              title: ascendant.details_title_s,
+              status: ascendant.details_state_s,
+              approvedDate: ascendant.system_originalApprovedDate_t
+            }
+          })
+        }
+      },
+      skip () { return !this.selectedDao?.docId },
+      variables () {
+        return {
+          id: this.proposalId,
+          daoId: this.selectedDao.docId
+        }
+      }
+    }
+  },
+
+  computed: {
+    ...mapGetters('dao', ['selectedDao'])
+  },
+
+  methods: {
+    parsedDate (date) {
+      return dateToStringShort(date)
+    },
+    cardColor (status) {
+      return {
+        'bg-primary': status === PROPOSAL_STATE.PROPOSED,
+        'bg-positive': status === PROPOSAL_STATE.APPROVED,
+        'bg-negative': status === PROPOSAL_STATE.REJECTED,
+        bordered: status === PROPOSAL_STATE.PENDING
+      }
+    },
+    iconColor (status) {
+      let color = ''
+      switch (status) {
+        case PROPOSAL_STATE.PROPOSED:
+          color = 'primary'
+          break
+        case PROPOSAL_STATE.APPROVED:
+          color = 'positive'
+          break
+        case PROPOSAL_STATE.REJECTED:
+          color = 'negative'
+          break
+        case PROPOSAL_STATE.PENDING:
+          color = 'white'
+          break
+      }
+      return color
+    },
+    cardStatus (card) {
+      let status = ''
+      if (card.approvedDate) {
+        status = this.parsedDate(card.approvedDate)
+      } else if (card.status === PROPOSAL_STATE.REJECTED) {
+        status = this.parsedDate(card.rejectedDate)
+      } else if (card.status === PROPOSAL_STATE.PROPOSED) {
+        status = 'On voting'
+      }
+      return status
+    },
+    completionCardTitle () {
+      let title = ''
+      if (this.progression.currentQuest.status === PROPOSAL_STATE.PROPOSED) {
+        title = 'Not available yet'
```

Comment:
@arsenijesavic you could just return the text here, so you wouldn't need the `title` variable. In that case you also would not need else, but you could just return `Create now!`